### PR TITLE
Wrap message's content into node's message

### PR DESCRIPTION
**Pull request by cecton**
Wraps the node's message content into another object so the ID of the connection is separated from the actual payload.

### CODE REVIEW

**Comment by maciejhirsz** [comment]
There should be a way to make it backwards compatible with a [`#[serde(untagged)]`](https://serde.rs/container-attrs.html#untagged) enum, I can look into it next week.

**Comment by dvdplm** [comment]
A PR description would be good.

FWIW I don't mind this as is, I like the explicitness of `V1`/`V2`.

**Comment by cecton** [comment]
(I keep it in draft because I'm still testing)

**Comment by cecton** [comment]
Ok I made a change to move the timestamp (ts) to the message.

I think this is better, you can clearly see the code in substrate-telemetry is easier. But there is something weird: `ts` wasn't use in the first place!! 😆  Should I remove it or do you plan to do something with it at some point?

I think this is better ALSO because for the code in substrate it removes some complexity for injecting the `ts` in the messages. But since it's not used at all... maybe we should consider removing it entirely?

Last but not least: if we do keep this new implementation I suggest I refactor the name of the struct Details to Payload to match the actual tag. But it's fine and it works as it is now.

Waiting for your feedback on this

**Comment by maciejhirsz** [comment]
> Should I remove it or do you plan to do something with it at some point?

I think it's fine to remove the timestamp, it will shave a bit of time parsing too. As long as it's being sent by the nodes, we can add it back in the future should we need it.

> Last but not least: if we do keep this new implementation I suggest I refactor the name of the struct Details to Payload to match the actual tag. But it's fine and it works as it is now.

Renaming sounds like a good idea to me.

**Comment by dvdplm** [comment]
> As long as it's being sent by the nodes, we can add it back in the future should we need it.

I think @cecton wants to remove it on the substrate side as well, right? If so, is that a problem for `substrate-telemetry`?

**Comment by maciejhirsz** [comment]
> > As long as it's being sent by the nodes, we can add it back in the future should we need it.
> 
> I think @cecton wants to remove it on the substrate side as well, right? If so, is that a problem for `substrate-telemetry`?

I don't think so, but it might be worth checking with `substrate-analytics` since it uses the same data as input AFAIU.

**Comment by dvdplm** [comment]
That is a really good point, pretty sure it's used there. K, so @cecton it's fine to remove it here but let's keep it in the outgoing payload.
cc @mattrutherford

**Comment by cecton** [comment]
I confirm there is indeed a ts field in substrate-analytics: https://github.com/paritytech/substrate-analytics/blob/master/src/web/root.rs#L251

But... the payload is completely different in V2. Would it still work with substrate-analytics if substrate nodes send a completely different payload?

**Comment by mattrutherford** [comment]
The timestamp is used in substrate-analytics and I think it could be useful in other cases too, so please if possible let's keep it in the substrate telemetry messages. If the data structure is different I can prepare a PR for that, no problem

**Comment by cecton** [comment]
OK I tested on current master and on my new branch and it works.

**Review by dvdplm — APPROVED** [review]
LGTM. 

It'd be good to have a test or two to prove backwards compatibility.

**Comment by cecton** [comment]
> It'd be good to have a test or two to prove backwards compatibility.

I think there were no test at all for this. I will add some for V1 and V2 then.

**Review by maciejhirsz — APPROVED** [review]
Looks good to me, a simple test with `serde_json::from_str` should suffice.

**Comment by cecton** [comment]
(I will wait a bit for the substrate part to be validated before merging in case there is a big change)

**Pull request merged**
